File: app/src/main/java/nl/rsdt/japp/jotial/data/bodies/LoginPostDataBuilder.java
package nl.rsdt.japp.jotial.data.bodies;

import com.google.gson.JsonObject;

import org.json.JSONObject;

import java.util.HashMap;

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 13-7-2016
 * Description...
 */
public class LoginPostDataBuilder {

    private String user;

    private String password;

    public LoginPostDataBuilder setUsername(String username) {
        this.user = username;
        return this;
    }

    public LoginPostDataBuilder setPassword(String password) {
        this.password = password;
        return this;
    }

    public String build()
    {
        JsonObject object = new JsonObject();
        object.addProperty("gebruiker", user);
        object.addProperty("ww", password);
        return object.toString();
    }

    public JSONObject buildAsParams() {
        HashMap<String, String> map =  new HashMap<>();
        map.put("gebruiker", user);
        map.put("ww", password);
        return new JSONObject(map);
    }

}

File: app/src/main/java/nl/rsdt/japp/jotial/data/bodies/LoginPostDataBuilder.kt
package nl.rsdt.japp.jotial.data.bodies

import com.google.gson.JsonObject
import org.json.JSONObject
import java.util.*

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 13-7-2016
 * Description...
 */
class LoginPostDataBuilder {

    private var user: String = ""

    private var password: String = ""

    fun setUsername(username: String): LoginPostDataBuilder {
        this.user = username
        return this
    }

    fun setPassword(password: String): LoginPostDataBuilder {
        this.password = password
        return this
    }

    fun build(): String {
        val `object` = JsonObject()
        `object`.addProperty("gebruiker", user)
        `object`.addProperty("ww", password)
        return `object`.toString()
    }

    fun buildAsParams(): JSONObject {
        val map = HashMap<String, String>()
        map["gebruiker"] = user
        map["ww"] = password
        return JSONObject(map)
    }

}
